yarn.lock

```diff
@@ -1344,6 +1344,23 @@
   resolved "https://registry.yarnpkg.com/@vue/shared/-/shared-3.0.8.tgz#b18d3a912299c0360c41bb42f03f502c5016bb1e"
   integrity sha512-zDdHjJjzgWYFw8TTUW08JjdERTlBik5/N5LuTAqU0e4wS6wElcvftiNHrD0ONr79O3fi66Z3Ug4jgPhBomPQUA==
 
+"@yelmor/vite-test-package-a@^1.0.0":
+  version "1.0.0"
+  resolved "https://registry.yarnpkg.com/@yelmor/vite-test-package-a/-/vite-test-package-a-1.0.0.tgz#8003fd876b6760adbdd8ec2777933e2728221264"
+  integrity sha512-Z1puRdOVVaZ+GdfdMA8EQdF+0vCTIFP/QbDm9egwNPH7E1y8j43XSJZxuDQDOLNMTrP2FGs+2rOyBlCr3wLCeg==
+
+"@yelmor/vite-test-package-a@^2.0.0":
+  version "2.0.0"
+  resolved "https://registry.yarnpkg.com/@yelmor/vite-test-package-a/-/vite-test-package-a-2.0.0.tgz#535c6a4c9351101ffdd04ca13ed48576b720816c"
+  integrity sha512-xuPg9ffq0ZH3G0jkb6dfcYKyN1xWpQp5/2JmOvWwxGXZkXBt4wC1px9v46OdLowwiqhySFB/DszRFMrVokw1WQ==
+
+"@yelmor/vite-test-package-b@^1.0.0":
+  version "1.0.0"
+  resolved "https://registry.yarnpkg.com/@yelmor/vite-test-package-b/-/vite-test-package-b-1.0.0.tgz#b3ea74a649a2f7bc70e9a1cd4bb05dcec955a41d"
+  integrity sha512-EKQK529cWMMPCoxFOOlINJXDPKpL8G8IwW7vk7vqjnkiP7Wx4B2sy9e+dd4xU86MlUSOlsNiEUwaH3HwdeGgOA==
+  dependencies:
+    "@yelmor/vite-test-package-a" "^1.0.0"
+
```

Comment:
Is this related to the PR?

Reply:
Yes.

It's not easy to make the test case.

I tryed to create test packages in the workspace, but there cannot be two different version of same package in workspace.

Then I use multiple lodash version as test package: 

```
├── node_modules
│   └── lodash
│       └── package.json    // v4.17.21 dependency of other packages
└── packages
    └── playground
        ├── nested-deps
        │   └── node_modules
        │       └── lodash
        │           └── package.json    // v4.0.0
        └── test-package-for-nested-deps
            └── node_modules
                └── lodash
                    └── package.json   // v3.0.0
```

It works fine when I run dev from the nested-deps directory, but since the test is run from workspace root, the `resolveDir` is workspace root too, resulting `lodash@4.17.21` being resolve and the test failed.

At last I have to use external test packages to construct a more realistic use case.

package a@1.0.0:
```
export default 'A@1.0.0'
```

package a@2.0.0:
```
export default 'A@2.0.0'
```

package b which depends on a@1.0.0:
```
export { default as A } from 'a'
export default 'B@1.0.0'
```

And to fulfill the scoped package problem in #3053, I used my own npm scope to publish the test packages.

Reply:
We need to find a solution that doesn't introduce foreign packages into our dependencies

Reply:
Well since we are testing dependency problem, we must have dependencies.

There is a hack solution that create test package in node_modules manually, but it is error-prone. It's not good to interfere with the package manager.

So it's better to use real packages installed from npm registry.

Guess the dependencies from personal scope looks doesn't public and secure,  would you please make some test packages under `@vite/` scope or something else?

Wish this solution is accpetable, and the test packages code listed above is simple and clear.

Reply:
@antfu

Reply:
Done.